solid/src/main/java/com/inrupt/client/solid/SolidContainer.java

```diff
@@ -100,14 +100,14 @@ public Set<SolidResource> getResources() {
 
     @Override
     public ValidationResult validate() {
-        // Get the normalized container URI
-        final String container = normalize(getIdentifier());
         final List<String> messages = new ArrayList<>();
         // Verify that the container URI path ends with a slash
-        if (!container.endsWith("/")) {
+        if (!getIdentifier().getPath().endsWith("/")) {
             messages.add("Container URI does not end in a slash");
         }
 
+        // Get the normalized container URI
+        final String container = normalize(getIdentifier());
```

Comment:
As above.

Maybe worth extracting something like `isContainer` for readability and consistency.

Reply:
done in https://github.com/inrupt/solid-client-java/pull/653/commits/2b4b91696bc4aa7f19774094d1f170183a16c9ac